website/content/docs/connect/manage-traffic/route-to-local-upstreams.mdx

```diff
@@ -83,23 +83,252 @@ You can configure the default routing behavior for all proxies in the mesh as we
-1. Apply the configuration by either calling the [`/config` HTTP API endpoint](/consul/api-docs/config) or running the [`consul config write` CLI command](/consul/commands/config/write). 
-   The following example writes a proxy defaults configuration entry from a local HCL file using the CLI:
+Create a service resolver configuration entry and specify the following fields:
+  - `Name`: The name of the target upstream service for which downstream clients should use locality-aware routing.
+  - `PrioritizeByLocality`: This block enables proxies in the mesh to use the region and zone defined in the service configuration to route traffic. Set the `mode` inside the block to `failover`. Refer to [`PrioritizeByLocality`](/consul/docs/connect/config-entries/service-resolver#prioritizebylocality) in the service resolver reference for details about the configuration.
+
+<Tabs>
+<Tab heading="HCL" group="hcl">
+<CodeBlockConfig filename="api-resolver.hcl">
 
-   ```shell-session
-   $ consul config write web-resolver.hcl
-   ```
+```hcl
+Kind     = "service-resolver"
+Name     = "api"
+PrioritizeByLocality = {
+    Mode = "failover"
+}
+```
+</CodeBlockConfig>
+</Tab>
+
+<Tab heading="JSON" group="json">
+<CodeBlockConfig filename="api-resolver.json">
+
+```json
+{
+  "kind": "service-resolver",
+  "name": "api",
+  "prioritizeByLocality": {
+    "mode": "failover"
+  }
+}
+```
+</CodeBlockConfig>
+</Tab>
+<Tab heading="Kubernetes" group="kubernetes">
+<CodeBlockConfig filename="api-resolver.yaml">
+
+```yaml
+apiversion: consul.hashicorp.com/v1alpha1
+kind: ServiceResolver
+metadata:
+  name: api
+spec:
+  prioritizeByLocality:
+    mode: failover
+```
+
+</CodeBlockConfig>
+</Tab>
+</Tabs>
+
+Apply the configuration by either running the [`consul config write` CLI command](/consul/commands/config/write), applying the Kubernetes CRD, or calling the [`/config` HTTP API endpoint](/consul/api-docs/config).
+
+<Tabs>
+<Tab heading="Consul CLI" group="cli">
+
+  ```shell-session
+  $ consul config write api-resolver.hcl
+  ```
+
+</Tab>
+
+<Tab heading="Kubernetes" group="k8s">
+
+  ```shell-session
+  $ kubectl apply -f api-resolver.yaml
+  ```
+
+</Tab>
+<Tab heading="HTTP API" group="api">
+
+  ```shell-session
+  $ curl --request PUT --data @api-resolver.hcl http://127.0.0.1:8500/v1/config
+  ```
+
+</Tab>
+</Tabs>
+
+### Explicit locality configuration on Kubernetes test clusters
+
+If you are testing this feature with a local Kubernetes cluster (or in an environment where `topology.kubernetes.io` labels are not set), it is possible to explicitly provide each node's locality using `kubectl label`:
+
+```shell-session
+kubectl label node $K8S_NODE topology.kubernetes.io/region="us-east-1" topology.kubernetes.io/zone="us-east-1a"
+```
+
+After setting these values, you should see subsequent service and proxy registrations in your cluster inherit the values from their local Kubernetes node.
+
```

Comment:
```suggestion
## Adjust load balancing and failover behavior

You can adjust the global or per-service load balancing and failover behaviors of routes between local service instances by apply the property override Envoy extension. The property override extension allows you to set and remove individual properties on the Envoy resources Consul generates. Refer to [Configure Envoy proxy properties](/consul/docs/connect/proxies/envoy-extensions/usage/property-override) for additional information.

1. Add the `EnvoyExtensions` configuration block to the service defaults or proxy defaults configuration entry.
1. Configure the following settings  in the `EnvoyExtensions` configuration:
   -  [`overprovisioning_factor`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint.proto#config-endpoint-v3-clusterloadassignment) 
   - [outlier detection](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#config-cluster-v3-outlierdetection) configuration. By default, Consul sets `overprovisioning_factor` to `100000`, which enforces total failover, and `max_ejection_percent` to `100`. Refer to the Envoy documentation about these fields before attempting to modify them.
1. Apply the configuration. Refer to [Apply the configuration entry](/consul/docs/connect/proxies/envoy-extensions/usage/property-override#apply-the-configuration-entry) for details. 
```
I wasn't sure how to write the steps for setting the overprovisioning factor and outlier detection, so that probably needs more detail. 

We should also consider just moving this section to the extension documentation and linking to it  from here. It seems to me that the actions we are describing are more a function of the extension than setting the locality for routes.

Reply:
> I wasn't sure how to write the steps for setting the overprovisioning factor and outlier detection, so that probably needs more detail.

I can help w/ that if we think an explicit example is needed, though [this](https://developer.hashicorp.com/consul/docs/connect/proxies/envoy-extensions/configuration/property-override#update-multiple-values-in-a-message-field) one is close already. Given this is even more into custom configuration territory, my thought was to maybe wait and see if we get requests for more detail as confirmation that it's something folks want to do. I could be wrong about that.

> We should also consider just moving this section to the extension documentation and linking to it from here. It seems to me that the actions we are describing are more a function of the extension than setting the locality for routes.

I could see that. My thinking was the extension is a general solution for potentially many needs, and the actions described are more specific to this feature, but use the extension to accomplish them. I'm open to adding an example there instead.

Reply:
Added your suggestion. I moved the default values callout to below the bullets since they apply to both properties.